mlir/test/mlir-tblgen/op-python-bindings.td

```diff
@@ -616,6 +616,15 @@ def VariadicRegionOp : TestOp<"variadic_region"> {
 // CHECK: def variadic_region(num_variadic, *, loc=None, ip=None)
 // CHECK:   return _get_op_result_or_op_results(VariadicRegionOp(num_variadic=num_variadic, loc=loc, ip=ip))
 
+// CHECK: @_ods_cext.register_operation(_Dialect)
+// CHECK: class WithSpecialCharactersOp(_ods_ir.OpView):
+// CHECK-LABEL: OPERATION_NAME = "test.123with- special.characters"
+def WithSpecialCharactersOp : TestOp<"123with- special.characters"> {
+}
+
+// CHECK: def _123with__special_characters(*, loc=None, ip=None)
```

Comment:
I can't see it right this moment but how come the `-` gets turned into two underscores? And is there a way to just have it be one?

Reply:
There is also a space in the operation name. I suppose that gets converted to an underscore, but I think we shouldn't have whitespace in the operation names.

Reply:
Definitely agree "no collapsing" and also secondly agree spaces shouldn't be allowed in mnemonics at all (I'm surprised they are...)